chore: max ai support for surveys and experiments

marandaneto opened the pull request:
## Problem

follow up https://github.com/PostHog/posthog/pull/37504

## Changes

<!-- If there are frontend changes, please include screenshots. -->
<!-- If a reference design was involved, include a link to the relevant Figma frame! -->

## How did you test this code?

<!-- Briefly describe the steps you took. -->
<!-- Include automated tests if possible, otherwise describe the manual testing routine. -->

<!-- Docs reminder: If this change requires updated docs, please do that! Engineers are the primary people responsible for their documentation. 🙌 -->

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

marandaneto committed:
chore: max ai support for surveys and experiments

marandaneto committed:
ref

marandaneto committed:
Merge branch 'master' into chore/max-ai-experiments

lucasheriques commented on frontend/src/scenes/experiments/ExperimentView/components.tsx:
```diff
@@ -58,6 +62,78 @@ import { getIndexForVariant } from '../legacyExperimentCalculations'
+                return
+            }
+            // Redirect to the new survey
+            router.actions.push(urls.survey(toolOutput.survey_id))
```
let's make sure to capture the product intent action of survey created here, similar to how we do it for empty states / survey form / max ai surveys, this way we can know if those features are being used succesfully too

lucasheriques commented on frontend/src/scenes/experiments/ExperimentView/components.tsx:
```diff
@@ -58,6 +62,78 @@ import { getIndexForVariant } from '../legacyExperimentCalculations'
+        },
+        callback: (toolOutput: { survey_id?: string; survey_name?: string; error?: string }) => {
+            if (toolOutput?.error || !toolOutput?.survey_id) {
+                posthog.captureException(toolOutput?.error, {
+                    source: 'survey-creation-failed',
+                    feature: 'surveys',
+                })
```
can reuse `captureMaxAISurveyCreationException` from surveys/utils.ts

maybe we can add a new optional parameter on it too, something like `context: "survey-experiments-upsell"` or something, so we know where it came from

lucasheriques reviewed (approved):
code looks good, left couple of comments, pls get back to me before merging but approving already as everything else is ✅

marandaneto committed:
Merge branch 'master' into chore/max-ai-experiments

marandaneto committed:
ref

marandaneto committed:
fix

marandaneto commented:
> code looks good, left couple of comments, pls get back to me before merging but approving already as everything else is ✅

addressed all comments, so i am merging it
